shopfloor_reception_packaging_dimension_mobile/static/src/scenario/reception_packaging_dimension.js

```diff
@@ -0,0 +1,144 @@
+        :record="state.data.packaging"
+        :options="{main: true, key_title: 'name', title_icon: 'mdi-package-variant'}"
+    />
+
+     <v-form ref="form_dimension">
+        <v-container>
+            <v-row>
+                <v-text-field
+                    label="Barcode"
+                    placeholder="Packaging Barcode"
+                    v-model="state.data.packaging.barcode"
+                ></v-text-field>
+            </v-row>
+            <v-row>
+                <v-text-field
+                    label="Quantiy"
+                    type="number"
+                    placeholder="Packaging Quantity"
+                    v-model="state.data.packaging.qty"
+                ></v-text-field>
+            </v-row>
+            <v-row>
+                <v-text-field
+                    label="Length"
+                    type="number"
+                    :suffix="state.data.packaging.length_uom"
+                    placeholder="Packaging Length"
+                    v-model="state.data.packaging.length"
+                ></v-text-field>
+            </v-row>
+            <v-row>
+                <v-text-field
+                    label="Width"
+                    type="number"
+                    :suffix="state.data.packaging.length_uom"
+                    placeholder="Packaging Width"
+                    v-model="state.data.packaging.width"
+                ></v-text-field>
+            </v-row>
+            <v-row>
+                <v-text-field
+                    label="Height"
+                    type="number"
+                    :suffix="state.data.packaging.length_uom"
+                    placeholder="Packaging Height"
+                    v-model="state.data.packaging.height"
+                ></v-text-field>
+            </v-row>
+            <v-row>
+                <v-text-field
+                    label="Max Weight"
+                    type="number"
+                    :suffix="state.data.packaging.weight_uom"
+                    placeholder="Packaging Max Weight"
+                    v-model="state.data.packaging.max_weight"
+                ></v-text-field>
+            </v-row>
+            <!-- extend -->
+       </v-container>
+    </v-form>
+
+    <div class="button-list button-vertical-list full">
+        <v-row align="center">
+            <v-col class="text-center" cols="12">
+                <btn-action action="todo" @click="state.on_done">Done</btn-action>
+            </v-col>
+        </v-row>
+        <v-row align="center">
+            <v-col class="text-center" cols="12">
+
+                <btn-action color="default" @click="state.on_skip">Skip</btn-action>
+
+            </v-col>
+        </v-row>
+    </div>
+</div>
+
+` +
+    template.substring(pos);
+
+// Extend the reception scenario with :
+//   - the new patched template
+//   - the js code for the new state
+const ReceptionPackageDimension = process_registry.extend("reception", {
```

Comment:
Can you try this?

```
Vue.componen("reception", {
     "methods": {
          [....]
      }
})
```

Reply:
2 reasons for this not to be working.
First one is in how we declare the Vue component for each scenario. At the moment those components are registered as `Anonymous Component`. But it can be fixed by adding the `name` key in the scenario component.

The 2nd one is at the time this file is being interpreted by the browser, the Vue component is not created yet, it is created later at the app creation https://github.com/OCA/wms/blob/eca1a4208830e96565016b7e204b307e10eb9539/shopfloor_mobile_base/static/wms/src/main.js#L65

Reply:
Yeah, you are right. I had an eureka moment some days ago in the shower: the main issue is that scenario components are registered as local components on the root app. That's it.
We should refactor this https://github.com/OCA/wms/blob/eca1a4208830e96565016b7e204b307e10eb9539/shopfloor_mobile_base/static/wms/src/main.js#L73 (I mean, drop it, and register such components in the std way).
Go ahead like this for now. Maybe add a TODO then we'll see.